py/src/braintrust/wrappers/openai.py

```diff
@@ -212,18 +213,29 @@ def on_span_start(self, span: tracing.Span[tracing.SpanData]) -> None:
             parent = self._spans[span.parent_id]
         else:
             parent = self._spans[span.trace_id]
-        self._spans[span.span_id] = parent.start_span(
+        created_span = parent.start_span(
             id=span.span_id,
             name=_span_name(span),
             type=_span_type(span),
             start_time=_timestamp_from_maybe_iso(span.started_at),
         )
+        self._spans[span.span_id] = created_span
+
+        # Enter the span context to make it active for current_span() calls
```

Comment:
I think we have a `Span.set_current()` method that does this without needing to bring in the context manager.